src/components/metadataSelects/useInitialOptionQuery.ts

```diff
@@ -0,0 +1,33 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { SelectOption } from '../../types'
+import { FilteredCategoryCombo } from './types'
+
+type InitialCategoryComboQueryResult = {
+    categoryCombo: FilteredCategoryCombo
+}
+
+export function useInitialOptionQuery({
+    selected,
+    onComplete,
+}: {
+    onComplete: (option: SelectOption) => void
+    selected?: string
+}) {
+    const INITIAL_OPTION_QUERY = {
+        categoryCombo: {
+            resource: `categoryCombos/${selected}`,
```

Comment:
I think we should use `id`-query prop here. Even though the result is the same, its more idiomatic imo.